src/pytorch_lightning/loggers/neptune.py

```diff
@@ -618,6 +619,9 @@ def _dict_paths(cls, d: dict, path_in_build: str = None) -> Generator:
     @property
     def name(self) -> str:
         """Return the experiment name or 'offline-name' when exp is run in offline mode."""
+        if self._run_name is None:
+            _ = self.run
+        assert self._run_name is not None
```

Comment:
@otaj 
The design here was intentional. As you can see in the base class, the name and version properties are Optional. This is because most loggers are not pickle friendly and we want to discourage accessing them in the main process before processes get spawned. This is why we create the experiments lazily, only when it is first needed. Hence, run name and run id would be None here because the experiment was not created yet.

Does it make sense?

Reply:
The idea makes sense. If the rest of the codebase is ok with`logger.name` being `None`, then so am I :joy: Thanks for the explanation.
I'm positive I saw a similar thing done in some other logger though, I'll have to look it up.

In the meantime, I'm reverting my two commits

Reply:
Thanks. I know it's suboptimal but it is the only design that generalized well across all loggers for multiprocessing. So far. I definitely agree it would be nice if we could have no optionals there.